tiledb/sm/query/readers/sparse_global_order_reader.cc

```diff
@@ -1651,15 +1651,18 @@ SparseGlobalOrderReader<BitmapType>::respect_copy_memory_budget(
         // Get the size for all tiles.
         uint64_t idx = 0;
         for (; idx < max_cs_idx; idx++) {
```

Comment:
Let's make a test to replicate the issue. Write values in a fragment. Write the exact same dimension values in another with different attribute values. Make a query condition that will clear out all values in the second fragment only. Read.